__tests__/ship.test.js

```diff
@@ -0,0 +1,293 @@
+                throw because _init is invoked on instantiation.
+            */
+            balticExplorer._isInitialized = false;
+            balticExplorer._currentPort = null;
+            balticExplorer._init();
+        });
+
+        afterEach(() => {
+            balticExplorer = null;
+        });
+
+        test('Throws an error if "_isInitialized" is already true', () => {
+            const error = 'The ship has already been initialized.';
+
+            expect(() => {
+                balticExplorer._init();
+            }).toThrowError(error);
+        });
+
+        test('State of "_currentPort" should NOT be null after _init', () => {
+            expect(balticExplorer._currentPort).not.toBeNull();
+        });
+
+        test('addShip should be called when instance is instantiatied', () => {
+            expect(balticExplorer._currentPort.addShip).toBeCalledWith(balticExplorer);
+        });
+
+        test('State of "_isInitialized" should be true after _init', () => {
+            expect(balticExplorer._isInitialized).toBe(true);
+        });
+
+        test('State of "_currentPort" should be null if "_itinerary.ports" is empty', () => {
+            const emptyConfig = {
+                _itinerary: {
+                    ports: []
+                }
+            };
+            const balticExplorer = new Ship(emptyConfig);
+            
+            expect(balticExplorer._currentPort).toBeNull();
+        });
+
+        test('State of "_isInitialized" should be false if "_itinerary.ports" is empty', () => {
+            const emptyConfig = {
+                _itinerary: {
+                    ports: []
+                }
+            };
+            const balticExplorer = new Ship(emptyConfig);
+
+            expect(balticExplorer._isInitialized).toBe(false);
+        });
+    });
+
+    describe('_setSail', () => {
+        let balticExplorerConfig;
+        let balticExplorer;
+        beforeEach(() => {
+            balticExplorerConfig = {
+                _itinerary: {
+                    ports: [
+                        {
+                            name: 'Helsinki',
+                            _ships: [],
+                            addShip: jest.fn(),
+                            removeShip: jest.fn()
+                        },
+                        {
+                            name: 'St Petersburg',
+                            _ships: [],
+                            addShip: jest.fn(),
+                            removeShip: jest.fn()
+                        }
+                    ]
+                }
+            };
+            balticExplorer = new Ship(balticExplorerConfig);
+        });
+
+        afterEach(() => {
+            balticExplorer = null;
+        });
+
+        test('Throws an error if "_itinerary.ports" is empty', () => {
+            const error = 'The ship has no itinerary.';
+
+            balticExplorer._itinerary.ports = [];
```

Comment:
Could add an extra line here to break up the act from the assert:
```suggestion
            balticExplorer._itinerary.ports = [];

```